lib/pbench/test/unit/server/query_apis/test_datasets_detail.py

```diff
@@ -12,69 +13,32 @@ class TestDatasetsDetail:
     constructor and `post` service.
     """
 
-    def build_index(self, server_config, dates):
-        """
-        Build the index list for query
-
-        Args:
-            dates (iterable): list of date strings
-        """
-        idx = server_config.get("Indexing", "index_prefix") + ".v6.run-data."
-        index = "/"
-        for d in dates:
-            index += f"{idx}{d},"
-        return index
-
-    def test_non_accessible_user_data(self, client, server_config, pbench_token):
-        """
-        Test behavior when Authorization header does not have access to other user's data
-        """
-        # The pbench_token fixture logs in as user "drb"
-        # Trying to access the data belong to the user "pp"
-        response = client.post(
-            f"{server_config.rest_uri}/datasets/detail",
-            headers={"Authorization": "Bearer " + pbench_token},
-            json={
-                "user": "pp",
-                "name": "dataset_name",
+    @pytest.fixture(autouse=True)
+    def _setup(self):
+        super()._setup(
+            pbench_endpoint="/datasets/detail",
+            elastic_endpoint="/_search?ignore_unavailable=true",
+            payload={
+                "user": "drb",
+                "name": "fio",
                 "start": "2020-08",
                 "end": "2020-10",
             },
+            bad_date_payload={
+                "user": "drb",
+                "name": "footest",
+                "start": "2020-12",
+                "end": "2020-19",
+            },
+            error_payload={"name": "foobar", "start": "2020-08", "end": "2020-08"},
+            empty_response_payload={
+                "hits": {
+                    "total": {"value": 0, "relation": "eq"},
+                    "max_score": None,
+                    "hits": [],
```

Comment:
Why is this value different from the one at line 24 in `lib/pbench/test/unit/server/query_apis/test_controllers_list.py`?

It seems like maybe it would be worth defining this value as a class attribute which could be used by code like this to initialize the instance attribute.  (That is, if the empty response payload is always the same, for those tests which specify it, then it would be good to have a common definition that those tests could use to specify it....)